src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1179,6 +1179,15 @@ public StringBuilder Append(char value)
                 return this;
             }
 
+            unsafe
+            {
+                Span<char> tempBuffer = stackalloc char[32]; // should be enough for all primitives
```

Comment:
```suggestion
                char* tempBuffer = stackalloc char[32]; // should be enough for all primitives
                if (value.TryFormat(new Span<char>(tempBuffer), out int charsWrittenToTmp, format: default, provider: null))
                {
                    return Append(tempBuffer, charsWrittenToTmp);
                }
```

Reply:
Why?

Reply:
> you need to pass a length to the span ctor that takes a pointer

yep, missed passing 32 there when typing in the comment box. Calling Append with Span calls into Append with char* so you are wrapping and unwrapping the same pointer with extra length checks.

Reply:
> with extra length checks.

Which provide safety.  Is there a huge performance difference this provides?  To get to this point, we just had to expand the size of the StringBuilder by allocating an entirely new StringBuilder under the covers.

Reply:
> Is there a huge performance difference this provides? 

no, it's only a suggestion for simpler implementation.

Reply:
Ok :) FWIW, I think not using unsafe and pointers leads to the implementation being simpler.

Reply:
I agree with Stephen. We strongly prefer safe code without pointers unless the unsafe code has a significant performance advantage.

Reply:
good to know :-)